modules/python/python-stats.c

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright (c) 2018 Viktor Tusa <tusavik@gmail.com>
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 as published
+ * by the Free Software Foundation, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * As an additional exemption you are allowed to compile & link against the
+ * OpenSSL libraries as published by the OpenSSL project. See the file
+ * COPYING for details.
+ *
+ */
+#include "python-stats.h"
+#include "stats/stats-registry.h"
+#include "stats/stats-cluster-single.h"
+#include <Python.h>
+
+typedef struct _PyStats
+{
+  PyObject_HEAD
+  StatsCounterItem *counter;
+  const char *stats_name;
+} PyStats;
+
+static PyTypeObject py_stats_type;
+
+PyObject *
+py_stats_new(PyTypeObject *type, PyObject *args, PyObject *kwds)
+{
+  PyStats *self = (PyStats *) type->tp_alloc(type, 0);
+
+  if (self != NULL)
+    {
+      if (!PyArg_ParseTuple(args, "s", &(self->stats_name)))
```

Comment:
I think you get a borrowed reference here and not an allocated string, e.g. youll need to strdup it.